src/main/java/io/appium/java_client/remote/MobileOptions.java

```diff
@@ -512,4 +518,20 @@ protected T amend(String optionName, Object value) {
         setCapability(optionName, value);
         return (T) this;
     }
+
+    @Override
+    public Map<String, Object> asMap() {
+        Map<String, Object> toReturn = new TreeMap<>();
+
+        Map<String, Object> capsMap = super.asMap();
+        capsMap.forEach((key, value) -> {
+            if (key.equals("platformName")) {
+                toReturn.put(key, value);
+            } else {
+                toReturn.put(APPIUM_PREFIX + key, value);
```

Comment:
should we also check if cap name is already prefixed?

Reply:
Yeah, I guess if we are allowing people to do `setCapability()` we should have some protection for that.